src/vim9instr.c

```diff
@@ -1823,6 +1826,23 @@ generate_CALL(
 		if (ufunc->uf_arg_types == NULL)
 		    continue;
 		expected = ufunc->uf_arg_types[i];
+
+		// class constructor with object member
+		if (class_constructor && expected->tt_type == VAR_ANY)
+		{
+		    class_T *clp = mtype->tt_class;
```

Comment:
Can you add a comment here describing what this block of code is doing?

Reply:
I have added a comment, but if there is an error or a better explanation, please change it:smile: